src/NHibernate/Transform/CacheableResultTransformer.cs

```diff
@@ -152,22 +172,23 @@ private CacheableResultTransformer(bool[] includeInTuple, bool[] includeInTransf
 			}
 		}
 
+		internal void SupplyAutoDiscoveredParameters(IResultTransformer transformer, string[] aliases)
+		{
+			if (!AutoDiscoverTypes)
+				throw new InvalidOperationException(
+					"Cannot supply auto-discovered parameters when it is not enabled on the transformer.");
+
+			var includeInTuple = ArrayHelper.Fill(true, aliases?.Length ?? 0);
+			InitializeTransformer(includeInTuple, GetIncludeInTransform(transformer, aliases, includeInTuple));
+		}
 
 		public object TransformTuple(object[] tuple, string[] aliases)
 		{
-			if (aliases != null && aliases.Length != _tupleLength)
-			{
-				throw new InvalidOperationException(
-					"aliases expected length is " + _tupleLength +
-					"; actual length is " + aliases.Length);
-			}
-			// really more correct to pass index( aliases.getClass(), aliases )
-			// as the 2nd arg to the following statement;
-			// passing null instead because it ends up being ignored.
```

Comment:
Test and comment on `aliases` here were quite obsolete, and the test was causing the auto-discovered cached query to fail. `aliases` are not used by the actual implementation, and the suggested one use a method which seems suppressed.